### PR TITLE
Transform and Extract pushed

### PR DESCRIPTION
**Pull request by BenBeard115**
Extract completed with functions moved around and multiprocessing.

### CODE REVIEW

File: transform.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
+
```
Comment (IshikaM19) [inline transform.py:1]: Maybe could add a docstring here :)

File: extract.py [hunk 2]
```diff
@@ -1,73 +1,25 @@
+from multiprocessing import Pool
```
Comment (IshikaM19) [inline extract.py:1]: Maybe could add a docstring :)

File: transform.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    return first_name, last_name, email, phone_number
+
+
+def make_dataframe(plants: list[list]) -> pd.DataFrame:
+    """Makes an empty dataframe with correct heading names"""
+    return pd.DataFrame(plants, columns=COLUMNS)
+
+
+def transform_main(plant_data: list[list[dict]]) -> pd.DataFrame:
+    with Pool(processes=4) as pool:
```
Comment (IshikaM19) [inline transform.py:101]: Maybe add a function doc string here :)

File: Pipeline/extract.py [hunk 4]
```diff
@@ -1,73 +1,25 @@
+from multiprocessing import Pool
```
Comment (IshikaM19) [inline Pipeline/extract.py:1]: Module Docstring would be good :)

File: Pipeline/transform.py [hunk 5]
```diff
@@ -0,0 +1,117 @@
+
```
Comment (IshikaM19) [inline Pipeline/transform.py:1]: Docstring here aswell would be good :)

**Review by clv1 — APPROVED** [review]
Approved - no changes beyond previous attempt to merge.

File: Pipeline/extract.py [hunk 6]
```diff
@@ -1,73 +1,25 @@
...
-        if response.get("error"):
-            continue
+def get_plant_data(plant_range: list[int]) -> list[dict]:
+    return [requests.get(f'{URL}{plant}').json() for plant in plant_range]
 
-        first_name, last_name, email, phone_number = get_botanist(response)
 
-        image_url = get_image(response)
+def extract_main():
+    """Multiprocessing of all plants and building their data into a dataframe"""
```
Comment (anukrr) [inline Pipeline/extract.py:12]: Could add typehints

File: Pipeline/transform.py [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+COLUMNS = ['plant_id', 'name', 'scientific_name', 'last_watered', 'recording_taken',
+           'soil_moisture', 'temperature', 'email', 'botanist_first_name',
+           'botanist_last_name', 'phone_number', 'image_url', 'longitude',
+           'latitude', 'town', 'country', 'country_abbreviation', 'continent']
+
+MIN_TEMP = 0
+MAX_TEMP = 25
+MIN_MOISTURE = 0
+MAX_MOISTURE = 100
+
```
Comment (anukrr) [inline Pipeline/transform.py:17]: nice variables

File: Pipeline/extract.py (hunk 4 above)
Comment (IshikaM19) [inline Pipeline/extract.py:1]: Sorry...I know I keep saying this...but worth putting a module docstring here 😂

**Review by anukrr — DISMISSED** [review]

File: Pipeline/transform.py [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+from multiprocessing import Pool
+import pandas as pd
+from extract import extract_main
+
+COLUMNS = ['plant_id', 'name', 'scientific_name', 'last_watered', 'recording_taken',
+           'soil_moisture', 'temperature', 'email', 'botanist_first_name',
+           'botanist_last_name', 'phone_number', 'image_url', 'longitude',
+           'latitude', 'town', 'country', 'country_abbreviation', 'continent']
+
+MIN_TEMP = 0
```
Comment (IshikaM19) [inline Pipeline/transform.py:13]: AMAZING!!!!

File: transform.py [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+            df = df._append(dataframe_segment, ignore_index=True)
+
+    return df
+
+
+if __name__ == "__main__":
+    plant_data = extract_main()
+    df = transform_main(plant_data)
+
+    print(df["temperature"])
```
Comment (IshikaM19) [inline transform.py:117]: Do we need this last print statement?

**Review by IshikaM19 — APPROVED** [review]

**Pull request merged**